rotors_gazebo_plugins/src/gazebo_ros_continous_joint_controller_plugin.cpp

```diff
@@ -0,0 +1,165 @@
+ *  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *  DISCLAIMED. IN NO EVENT SHALL Antons Rebguns <email> BE LIABLE FOR ANY
+ *  DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ *  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ *  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ *THIS
+ *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+#include <math.h>
+#include <boost/bind.hpp>
+#include <boost/thread/mutex.hpp>
+#include <sdf/sdf.hh>
+#include "rotors_comm/SetFrequency.h"
+#include "rotors_gazebo_plugins/common.h"
+#include "rotors_gazebo_plugins/motor_state.h"
+#include "rotors_gazebo_plugins/gazebo_ros_continous_joint_controller_plugin.h"
+
+namespace gazebo {
+
+ContinousJointControllerPlugin::ContinousJointControllerPlugin()
+    : alive_(true) {}
+
+// Destructor
+ContinousJointControllerPlugin::~ContinousJointControllerPlugin() {
+  delete nh_;
+}
+
+// Load the controller
+void ContinousJointControllerPlugin::Load(physics::ModelPtr parent,
+                                          sdf::ElementPtr sdf) {
+  this->parent_ = parent;
+  this->world_ = parent->GetWorld();
+  getSdfParam<std::string>(sdf, "robot_namespace", robot_namespace_, "");
+
+  joint_ = GetReferencedJoint(parent, sdf, "joint");
+
+  if (joint_ == nullptr) {
+    ROS_FATAL("No joint was found");
+    return;
+  }
+
+  // Make sure the ROS node for Gazebo has already been initialized
+  if (!ros::isInitialized()) {
+    ROS_FATAL_STREAM(
+        "A ROS node for Gazebo has not been initialized, unable to load "
+        "plugin. "
+        << "Load the Gazebo system plugin 'libgazebo_ros_api_plugin.so' in the "
+           "gazebo_ros package)");
+    return;
+  }
+
+  nh_ = new ros::NodeHandle(this->robot_namespace_);
+
+  // no need for position control here
+  current_motor_state_.mode = MotorStateMode::Velocity;
+  current_motor_state_.torque_enabled = true;
+
+  // read sdf parameters
+  getSdfParam<double>(sdf, "default_pos", current_motor_state_.goal_pos_rad,
+                      0.0);
+  current_motor_state_.current_pos_rad = current_motor_state_.goal_pos_rad;
+  getSdfParam<double>(sdf, "default_velocity_limit",
+                      current_motor_state_.velocity_limit_rad_s, 100.0);
+  getSdfParam<double>(sdf, "default_torque_limit",
+                      current_motor_state_.torque_limit, 10);
+  getSdfParam<std::string>(sdf, "frequency_service_name", service_name_,
+                           "set_frequency");
+
+  std::string joint_state_topic_name;
+  getSdfParam<std::string>(sdf, "joint_state_topic_name",
+                           joint_state_topic_name, "joint_state");
+  joint_->SetPosition(0, current_motor_state_.current_pos_rad);
```

Comment:
SetPosition, GetParam, SetParam seem to be incompatible with gazebo 2, that's why Jenkins fail on ubuntu trusty.
`trusty/src/rotors_simulator/rotors_gazebo_plugins/src/gazebo_ros_continous_joint_controller_plugin.cpp: In member function 'virtual void gazebo::ContinousJointControllerPlugin::Load(gazebo::physics::ModelPtr, sdf::ElementPtr)':
/home/jenkins/workspace/rotors_simulator/osx/ubuntu-trusty/src/rotors_simulator/rotors_gazebo_plugins/src/gazebo_ros_continous_joint_controller_plugin.cpp:92:11: error: 'class gazebo::physics::Joint' has no member named 'SetPosition'
   joint_->SetPosition(0, current_motor_state_.current_pos_rad);
           ^
/home/jenkins/workspace/rotors_simulator/osx/ubuntu-trusty/src/rotors_simulator/rotors_gazebo_plugins/src/gazebo_ros_continous_joint_controller_plugin.cpp: In member function 'void gazebo::ContinousJointControllerPlugin::UpdateMotor(const MotorState&)':
/home/jenkins/workspace/rotors_simulator/osx/ubuntu-trusty/src/rotors_simulator/rotors_gazebo_plugins/src/gazebo_ros_continous_joint_controller_plugin.cpp:139:15: error: 'class gazebo::physics::Joint' has no member named 'GetParam'
   if (joint_->GetParam("vel", 0) != current_motor_state_.velocity_rad_s) {
               ^
/home/jenkins/workspace/rotors_simulator/osx/ubuntu-trusty/src/rotors_simulator/rotors_gazebo_plugins/src/gazebo_ros_continous_joint_controller_plugin.cpp:140:13: error: 'class gazebo::physics::Joint' has no member named 'SetParam'
     joint_->SetParam("vel", 0, current_motor_state_.velocity_rad_s);
             ^
/home/jenkins/workspace/rotors_simulator/osx/ubuntu-trusty/src/rotors_simulator/rotors_gazebo_plugins/src/gazebo_ros_continous_joint_controller_plugin.cpp:145:13: error: 'class gazebo::physics::Joint' has no member named 'SetParam'
     joint_->SetParam("fmax", 0, read_motor_state.torque_limit);
             ^
/home/jenkins/workspace/rotors_simulator/osx/ubuntu-trusty/src/rotors_simulator/rotors_gazebo_plugins/src/gazebo_ros_continous_joint_controller_plugin.cpp:147:13: error: 'class gazebo::physics::Joint' has no member named 'SetParam'
     joint_->SetParam("fmax", 0, 0.0);`

Reply:
Thanks, it got farther than it did last time. I updated the CMakeLists to skip this plugin if the gazebo version is too low